manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1062,454 @@ def text2svg(self):
+        tags = re.finditer(
+            '<gradient\s+from="([^"]+)"\s+to="([^"]+)"(\s+offset="([^"]+)")?>(.+?)</gradient>',
+            self.original_text,
+            re.S,
+        )
+        gradientmap = []
+        for tag in tags:
+            start = self._count_real_chars(self.original_text[: tag.start(0)])
+            end = start + self._count_real_chars(tag.group(5))
+            offsets = tag.group(4).split(",") if tag.group(4) else [0]
+            start_offset = int(offsets[0]) if offsets[0] else 0
+            end_offset = int(offsets[1]) if len(offsets) == 2 and offsets[1] else 0
+
+            gradientmap.append(
+                {
+                    "start": start,
+                    "end": end,
+                    "from": tag.group(1),
+                    "to": tag.group(2),
+                    "start_offset": start_offset,
+                    "end_offset": end_offset,
+                }
+            )
+        self.text = re.sub("<gradient[^>]+>(.+?)</gradient>", r"\1", self.text, 0, re.S)
+        return gradientmap
+
+    def _parse_color(self, col):
+        """Internally used function.
+        Parse color given in ``<color>`` or ``<gradient>`` tags."""
+        if re.match("#[0-9a-f]{6}", col):
+            return col
+        else:
+            return Colors[col.lower()].value
+
+    def extract_color_tags(self):
+        """Internally used function.
+        Used to determine what parts (if any) of the string should be formatted with a custom color.
+        Removes the ``<color>`` tag, as it is not part of Pango's markup and would cause an error."""
+        tags = re.finditer(
+            '<color\s+col="([^"]+)"(\s+offset="([^"]+)")?>(.+?)</color>',
+            self.original_text,
+            re.S,
+        )
+
+        colormap = []
+        for tag in tags:
+            start = self._count_real_chars(self.original_text[: tag.start(0)])
+            end = start + self._count_real_chars(tag.group(4))
+            offsets = tag.group(3).split(",") if tag.group(3) else [0]
+            start_offset = int(offsets[0]) if offsets[0] else 0
+            end_offset = int(offsets[1]) if len(offsets) == 2 and offsets[1] else 0
+
+            colormap.append(
+                {
+                    "start": start,
+                    "end": end,
+                    "color": tag.group(1),
+                    "start_offset": start_offset,
+                    "end_offset": end_offset,
+                }
+            )
+        self.text = re.sub("<color[^>]+>(.+?)</color>", r"\1", self.text, 0, re.S)
+        return colormap
+
+    def __repr__(self):
+        return f"MarkupText({repr(self.original_text)})"
+
+    def remove_last_M(self, file_name):
+        with open(file_name, "r") as fpr:
+            content = fpr.read()
+        content = re.sub(r'Z M [^A-Za-z]*? "\/>', 'Z "/>', content)
+        with open(file_name, "w") as fpw:
+            fpw.write(content)
```

Comment:
possibly define this also as a static method.

Reply:
I would then move that to the `PangoUtils`, is that OK for you?